Solutions/Marain.UserNotifications.Abstractions/Marain/UserNotifications/UserNotificationMetadata.cs

```diff
@@ -16,19 +19,19 @@ public class UserNotificationMetadata
         /// </summary>
         /// <param name="correlationIds">The <see cref="CorrelationIds"/>.</param>
         /// <param name="etag">The <see cref="ETag"/>.</param>
-        public UserNotificationMetadata(string[] correlationIds, string? etag)
+        public UserNotificationMetadata(IEnumerable<string> correlationIds, string? etag)
         {
             // Note: although correlation Ids should not be null, some combinations of serializer settings can result
             // in an empty array not being serialized, meaning that when this constructor is called during
             // deserialization we will get a null value. As such, we'll just treat null the same as an empty array.
-            this.CorrelationIds = correlationIds ?? Array.Empty<string>();
+            this.CorrelationIds = (correlationIds ?? Enumerable.Empty<string>()).ToImmutableArray();
```

Comment:
You've not actually declared `correlationIds` as nullable here, so strictly speaking we should throw `ArgumentNullException` on the right of the `??`. However, if you meant to make that nullable, then the same comments as above applied—creating an empty enumerable and then converting it to an empty immutable array goes slightly round the houses.